### DISCUSSION TITLE
Tasmotised MS-105 Dimmers Beeping on Wi-Fi Disconnect

### DISCUSSION THREAD

**Discussion by Teleportist** [arendst/Tasmota#12825] [Q&A] [answered]
I have 4 of [these](https://templates.blakadder.com/moes-MS-105-1.html), flashed with Tasmota 8.1.0.2.
All work great, but if my wireless router ever resets, they all start repeatedly beeping, because they've lost connection.
A problem if this happens in the middle of the night!

I thought the buzzers might be somehow linked to LedPower, so tried the command '_Backlog LedPower 0; SetOption31 1_' in console. Following with '_ledpower_' confirms the LED is now off.
But the issue remains.

Is there any way to disable the beeping please?
It's not practical for me to access all 4 dimmers, open them and try to disconnect/de-solder the tone generator.

Thanks.

**Comment by ascillato2** [comment]
You will need to compile by yourself undefining `USE_BUZZER`

https://github.com/arendst/Tasmota/blob/development/BUILDS.md

https://tasmota.github.io/docs/Compile-your-build/

**Reply by Teleportist** [reply]
Thanks. But isn’t there a Tasmota command I can issue in console to disable the buzzer?

**Reply by ascillato** [reply]
No, sorry. There isn't.

**Reply by barbudor** [reply]
@Teleportist are you using the template that you referred in your 1st post : https://templates.blakadder.com/moes-MS-105-1.html

In this template, neither the Buzzer nor any led is under control of Tasmota. All is under the control of the TuyaMCU.
So the decision of buzzing seems to be a decision of the TuyaMCU itself.
From what I see in the code, Tasmota let the TuyaMCU knows that the Wifi is gone which then in turns decide to buzz.

To confirm, can you set WebLog 3 and simulate a Wifi disconnection ?

**Reply by Teleportist** [reply]
Thanks. What is the best way to simulate a wifi disconnection and how can I access the weblog please?

**Reply by barbudor** [reply]
Weblog is the Tasmota web console accessible from the web gui
You have to type the command `weblog 3` there
To disconnect the wifi, unplug the power of you Access Point ;)

> if my wireless router ever resets

So you know how to do it

**Comment by Teleportist** [comment]
Thanks. I was hoping I wouldn’t have to unplug my access point…

How to I view/download the weblog once the device reconnects? Will it persist in the console?

…or taking another tack, is there a way to alter the template to take control and disable the buzzer?

**Reply by barbudor** [reply]
> How to I view/download the weblog once the device reconnects? Will it persist in the console?

It should. No need to cut it long. It should start buzzing, then replug.

> …or taking another tack, is there a way to alter the template to take control and disable the buzzer?

You didn't seem to have understood what I wrote above.
There is no way to alter the template because Tasmota do not control the Buzzer.
Only the TuyaMCU does.

Once we can confirm that the TuyaMCU start buzzing because Tasmota tell him that the wifi is gone, we could look at how to prevent this.

**Reply by sfromis** [reply]
You could also physically rewire the dimmer to connect the buzzer to ESP8266 instead of to the chip running TuyaMCU.

**Reply by barbudor** [reply]
> You could also physically rewire the dimmer to connect the buzzer to ESP8266 instead of to the chip running TuyaMCU.

He wants to avoid re-opening the walls to get the units back on table.
As far as he gets the units on the table, then totally removing the buzzer would be the way to shut it down

**Comment by Teleportist** [comment]
OK. Here's the console output after the AP reset.

00:00:00 TYA: Send dim skipped value=0
00:00:00 Project tasmota Dimmer-1 Version 8.1.0.2(tasmota)-2_6_1
00:00:00 SNS: Hardware Serial
00:00:00 TYA: Request MCU configuration
00:00:00 TYA: Send "55aa0001000000"
00:00:00 TYA: Set WiFi LED 2 (-1)
00:00:00 TYA: Send "55aa000300010205"
00:00:00 {"TuyaReceived":{"Data":"55AA0003000002","Cmnd":3}}
00:00:00 TYA: RX WiFi LED set ACK
00:00:00 WIF: Attempting connection...
00:00:00 WIF: Connecting to AP1 ********* in mode 11N as Dimmer-1...
00:00:01 WIF: Attempting connection...
00:00:02 WIF: Attempting connection...
00:00:04 WIF: Attempting connection...
00:00:05 WIF: Connected
00:00:05 HTP: Web server active on Dimmer-1 with IP address 192.168.***.***
00:00:05 TYA: Set WiFi LED 3 (0)
00:00:05 TYA: Send "55aa000300010306"
00:00:05 {"TuyaReceived":{"Data":"55AA0003000002","Cmnd":3}}
00:00:05 TYA: RX WiFi LED set ACK
12:49:45 MQT: Attempting connection...
12:49:45 MQT: Connected
12:49:45 MQT: tele/Dimmer_1/LWT = Online (retained)
12:49:45 MQT: cmnd/Dimmer_1/POWER = 
12:49:45 MQT: Subscribe to cmnd/Dimmer_1/#
12:49:45 MQT: Subscribe to cmnd/tasmotas/#
12:49:45 MQT: Subscribe to cmnd/Dimmer_1_fb/#
12:49:45 MQT: tele/Dimmer_1/INFO1 = {"Module":"MS-105","Version":"8.1.0.2(tasmota)","FallbackTopic":"cmnd/Dimmer_1_fb/","GroupTopic":"cmnd/tasmotas/"}
12:49:45 MQT: tele/Dimmer_1/INFO2 = {"WebServerMode":"Admin","Hostname":"Dimmer-1","IPAddress":"192.168.***.***"}
12:49:45 MQT: tele/Dimmer_1/INFO3 = {"RestartReason":"Software/System restart"}
12:49:45 MQT: stat/Dimmer_1/RESULT = {"POWER":"OFF"}
12:49:45 MQT: stat/Dimmer_1/POWER = OFF
12:49:45 NTP: Drift 0, (UTC) Thu Sep 02 11:49:45 2021, (DST) Sun Mar 28 02:00:00 2021, (STD) Sun Oct 31 03:00:00 2021
12:49:45 QPC: Reset
12:49:45 TYA: Set WiFi LED 4 (0)
12:49:45 TYA: Send "55aa000300010407"
12:49:45 {"TuyaReceived":{"Data":"55AA0003000002","Cmnd":3}}
12:49:45 TYA: RX WiFi LED set ACK
12:49:47 APP: Boot Count 35
12:49:47 CFG: Saved to flash at F4, Count 506, Bytes 4096
12:49:48 TYA: Send "55aa00000000ff"
12:49:48 {"TuyaReceived":{"Data":"55AA000000010101","Cmnd":0,"CmndData":"01"}}
12:49:48 TYA: Heartbeat
12:49:49 MQT: tele/Dimmer_1/STATE = {"Time":"2021-09-02T12:49:49","Uptime":"0T00:00:12","UptimeSec":12,"Heap":32,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"OFF","Dimmer":22,"Fade":"OFF","Speed":1,"LedTable":"ON","Wifi":{"AP":1,"SSId":"*********","BSSId":"**:**:**:**:**:**","Channel":6,"RSSI":44,"Signal":-78,"LinkCount":1,"Downtime":"0T00:00:06"}}
12:49:59 TYA: Send "55aa00000000ff"
12:49:59 {"TuyaReceived":{"Data":"55AA000000010101","Cmnd":0,"CmndData":"01"}}
12:49:59 TYA: Heartbeat

**Reply by barbudor** [reply]
I was expecting a log starting from Tasmota connected to the Wifi/MQTT, then loosing Wifi, then connecting again ...
Where in this log is the wifi loss event ?
This all looks like a restart of the Tasmota device, not of the AP.

**Reply by barbudor** [reply]
Whatever,
I see that there is a message that Tasmota is sending to TuyaMCU to set the WifiLed
There is no message specific to the buzzer
So the only solution I can see is disabling in Tasmota code the sending the message that the Wifi is gone.
Which means that any Wifi led will remains active
Which I don't see that as a problem for a device that inside the wall

**Reply by Teleportist** [reply]
When the wifi was lost the dimmer dropped back to access point mode for 180 seconds, then automatically restarted and looked for the router again. This is why the log is missing the disconnection event, and is what I was concerned may happen.

If Tasmota can control the wifi LED is there any reason why it can’t therefore also control the buzzer?

Otherwise, what would be the procedure to disable the sending to Tuya of the wifi disconnection message in the Tasmota code please?

**Reply by barbudor** [reply]
1) with command `WifiConfig 5` you can prevent Tasmota to restart and just wait for the AP to come back. Try it, it may help

2) again and again, on this device Tasmota controls NOTHING, neither led, buzzer or dimmer
It just pass commands to TuyaMCU
It is TuyaMCu that decide to buzz when tasmota tells it about wifi loss

**Comment by Teleportist** [comment]
Thanks again.

So I set the wifi config 5 and here the full log;

08:35:04 CMD: weblog 3
08:35:04 MQT: stat/Dimmer_1/RESULT = {"WebLog":3}
08:35:04 CFG: Saved to flash at F9, Count 509, Bytes 4096
08:35:14 TYA: Send "55aa00000000ff"
08:35:14 {"TuyaReceived":{"Data":"55AA000000010101","Cmnd":0,"CmndData":"01"}}
08:35:14 TYA: Heartbeat
08:35:25 TYA: Send "55aa00000000ff"
08:35:25 {"TuyaReceived":{"Data":"55AA000000010101","Cmnd":0,"CmndData":"01"}}
08:35:25 TYA: Heartbeat
08:35:36 TYA: Send "55aa00000000ff"
08:35:36 {"TuyaReceived":{"Data":"55AA000000010101","Cmnd":0,"CmndData":"01"}}
08:35:36 TYA: Heartbeat
08:35:47 TYA: Send "55aa00000000ff"
08:35:47 {"TuyaReceived":{"Data":"55AA000000010101","Cmnd":0,"CmndData":"01"}}
08:35:47 TYA: Heartbeat
08:35:58 TYA: Send "55aa00000000ff"
08:35:58 {"TuyaReceived":{"Data":"55AA000000010101","Cmnd":0,"CmndData":"01"}}
08:35:58 TYA: Heartbeat
08:36:09 TYA: Send "55aa00000000ff"
08:36:09 {"TuyaReceived":{"Data":"55AA000000010101","Cmnd":0,"CmndData":"01"}}
08:36:09 TYA: Heartbeat
08:36:20 TYA: Send "55aa00000000ff"
08:36:20 {"TuyaReceived":{"Data":"55AA000000010101","Cmnd":0,"CmndData":"01"}}
08:36:20 TYA: Heartbeat
08:41:29 WIF: Connect failed with AP incorrect password
08:41:30 WIF: Connect failed with AP incorrect password
08:41:31 WIF: Connect failed with AP incorrect password
08:41:31 TYA: Send "55aa00000000ff"
08:41:31 {"TuyaReceived":{"Data":"55AA000000010101","Cmnd":0,"CmndData":"01"}}
08:41:31 TYA: Heartbeat
08:41:32 WIF: Connect failed with AP incorrect password
08:41:33 WIF: Connect failed with AP incorrect password
08:41:34 WIF: Connect failed with AP incorrect password
08:41:35 WIF: Connect failed with AP incorrect password
08:41:36 WIF: Connect failed with AP incorrect password
08:41:37 WIF: Connect failed with AP incorrect password
08:41:38 WIF: Connect failed with AP incorrect password
08:41:39 WIF: Connect failed with AP incorrect password
08:41:40 WIF: Connect failed with AP incorrect password
08:41:41 WIF: Connect failed with AP incorrect password
08:41:42 WIF: Connect failed with AP incorrect password
08:41:42 TYA: Send "55aa00000000ff"
08:41:42 {"TuyaReceived":{"Data":"55AA000000010101","Cmnd":0,"CmndData":"01"}}
08:41:42 TYA: Heartbeat
08:41:43 WIF: Connect failed with AP incorrect password
08:41:44 WIF: Connect failed with AP incorrect password
08:41:45 WIF: Connect failed with AP incorrect password
08:41:46 WIF: Connect failed with AP incorrect password
08:41:47 WIF: Connect failed with AP incorrect password
08:41:48 WIF: Connect failed with AP incorrect password
08:41:49 WIF: Connect failed with AP incorrect password
08:41:50 WIF: Connect failed with AP incorrect password
08:41:51 WIF: Connect failed with AP incorrect password
08:41:52 WIF: Connect failed with AP incorrect password
08:41:53 WIF: Connect failed with AP incorrect password
08:41:53 TYA: Send "55aa00000000ff"
08:41:53 {"TuyaReceived":{"Data":"55AA000000010101","Cmnd":0,"CmndData":"01"}}
08:41:53 TYA: Heartbeat
08:41:54 WIF: Connect failed with AP incorrect password
08:41:55 WIF: Connect failed with AP incorrect password
08:41:56 WIF: Connect failed with AP incorrect password
08:41:57 WIF: Connect failed with AP incorrect password
08:41:58 WIF: Connect failed with AP incorrect password
08:41:59 WIF: Connected
08:41:59 TYA: Set WiFi LED 3 (0)
08:41:59 TYA: Send "55aa000300010306"
08:41:59 {"TuyaReceived":{"Data":"55AA0003000002","Cmnd":3}}
08:41:59 TYA: RX WiFi LED set ACK
08:42:04 TYA: Send "55aa00000000ff"
08:42:04 {"TuyaReceived":{"Data":"55AA000000010101","Cmnd":0,"CmndData":"01"}}
08:42:04 TYA: Heartbeat
08:42:08 MQT: Attempting connection...
08:42:08 MQT: Connected
08:42:08 MQT: tele/Dimmer_1/LWT = Online (retained)
08:42:08 MQT: cmnd/Dimmer_1/POWER = 
08:42:08 MQT: Subscribe to cmnd/Dimmer_1/#
08:42:08 MQT: Subscribe to cmnd/tasmotas/#
08:42:08 MQT: Subscribe to cmnd/Dimmer_1_fb/#
08:42:08 TYA: Set WiFi LED 4 (0)
08:42:08 TYA: Send "55aa000300010407"
08:42:08 {"TuyaReceived":{"Data":"55AA0003000002","Cmnd":3}}
08:42:08 TYA: RX WiFi LED set ACK
08:42:15 TYA: Send "55aa00000000ff"
08:42:15 {"TuyaReceived":{"Data":"55AA000000010101","Cmnd":0,"CmndData":"01"}}
08:42:15 TYA: Heartbeat

**Reply by barbudor** [reply]
Could you let me know if you heared the buzzer while Wifi was disconnected ?
During which time ?
It should have been between
`08:41:29 WIF: Connect failed with AP incorrect password`
and
`08:41:59 WIF: Connected`
but I can see any Tuya message here that would tell the TuyaMCU that the Wifi is gone

The only related message are:
```
08:41:59 TYA: Set WiFi LED 3 (0)           ==> 3: WIFI_RESTART
08:41:59 TYA: Send "55aa000300010306"
```
quickly followed with
```
08:42:08 TYA: Set WiFi LED 4 (0)           => 4: WIFI_CONNECTED but sent if MQTT is connected
08:42:08 TYA: Send "55aa000300010407"
```
But that happens right at the end.

**Reply by Teleportist** [reply]
The buzzer sounded as soon as the wifi dropped, then continued until reconnection.

It's difficult to tell exact times, as I've 4 of these devices in one room, each dropping and reconnecting at slightly different times.

Thanks

**Reply by Teleportist** [reply]
Any way to stop Tasmota sending the first message? Maybe that would stop the buzzer sounding?

Or else send the second message straight after, so I still get notification there's a connection issue, but ihe busser only sounds a couple of times?

**Reply by barbudor** [reply]
I just noticed that you are running an old version.
Although the code in this region of the TuyaMCU module seems to be the same, the management of Wifi has been widely changed.
Could you please check a more recent version like 9.5.0 or the latest development version ?

Are you using an official binary or a custom compiled binary ?

**Reply by Teleportist** [reply]
I can upgrade one of the dimmers to 9.5.0 sure. Will do and report back.

Current software is just the official bin.

**Comment by Teleportist** [comment, marked as answer]
OK, so I've now upgraded one of the buzzers to 9.5.0 and disconnected again.

I also set all 4 dimmers to wificonfig 5 'wait'. No buzzing!!!
I've posted the log of the updated dimmer for comparison below, but it seems Tasmota
is effecting whether the unit buzzes or not, based on wificonfig.

So for now I'll leave all units set to wificonfig 5, but it would give peace of mind if this could
be addressed, to enable wificonfig 2 (wifi manager) to be used again. Just in case.

09:10:48.358 CFG: Saved to flash at FA, Count 525, Bytes 4096
09:10:57.782 MQT: Attempting connection...
09:10:57.852 MQT: Connected
09:10:57.856 MQT: tele/Dimmer_1/LWT = Online (retained)
09:10:57.859 MQT: cmnd/Dimmer_1/POWER =
09:10:57.862 MQT: Subscribe to cmnd/Dimmer_1/#
09:10:57.866 MQT: Subscribe to cmnd/tasmotas/#
09:10:57.869 MQT: Subscribe to cmnd/Dimmer_1_fb/#
09:10:57.874 MQT: Unsubscribe from homeassistant/status
09:10:58.039 TYA: Set WiFi LED 4 (0)
09:10:58.050 TYA: Send "55aa000300010407"
09:10:58.059 TYA: Send "55aa00000000ff"
09:10:58.063 {"TuyaReceived":{"Data":"55AA0003000002","Cmnd":3}}
09:10:58.065 TYA: RX WiFi LED set ACK
09:10:58.154 {"TuyaReceived":{"Data":"55AA000000010101","Cmnd":0,"CmndData":"01"}}
09:10:58.157 TYA: Heartbeat
09:11:09.036 TYA: Send "55aa00000000ff"
09:11:09.046 {"TuyaReceived":{"Data":"55AA000000010101","Cmnd":0,"CmndData":"01"}}

Many thanks.
